src/Tests/Microsoft.DotNet.PackageValidation.Tests/ValidatePackageTargetTests.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+using Microsoft.NET.TestFramework;
+using Microsoft.NET.TestFramework.Commands;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.DotNet.PackageValidation.Tests
+{
+    public class ValidatePackageTargetTests : SdkTest
+    {
+        public ValidatePackageTargetTests(ITestOutputHelper log) : base(log)
+        {
+            // clearing the cache
+            string localSdkNugetCacheValidationPath = Path.Combine(TestContext.Current.NuGetCachePath, "microsoft.dotnet.packagevalidation");
+            if (Directory.Exists(localSdkNugetCacheValidationPath))
+                Directory.Delete(localSdkNugetCacheValidationPath, recursive: true);
+
+            // Packing and copying the local version package validation
+            string packageValidationProjectPath = Path.Combine(Path.GetDirectoryName(_testAssetsManager.TestAssetsRoot), "Compatibility", "Microsoft.DotNet.PackageValidation", "Microsoft.DotNet.PackageValidation.csproj");
```

Comment:
Should we be concerned that there might be locked files and that might cause the test to fail? Or is this the pattern that the SDK tests use?

Reply:
sdk doesnt produce any packages so there is no precedent here, but yes there could be a race condition here. i will add the lock.

Reply:
We shouldn't be polluting the nuget cache with build output, nor should we be deleting things from the nuget cache.  Come up with a pattern to import the props/targets as if the actual package was built and installed.

Reply:
> We shouldn't be polluting the nuget cache with build output, nor should we be deleting things from the nuget cache.

this is not the machine build output. its specific to the repo. eg the path is ```C:\git\sdk\artifacts\.nuget\packages```

Reply:
This repo produces an SDK.  Also, consumption via package isn’t the mainline scenario. We want this to be part of the .NET SDK, right?  Can’t we leverage the existing hooks that the SDK uses to test itself?  @dsplaisted

Reply:
I think I was under the impression that initially this would be consumed as a package.  If it's already going in the .NET SDK, then yes, we should just test it as built-in functionality.